src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp

```diff
@@ -283,26 +291,25 @@ GetFeatures(const ProblemDescription& problem, std::size_t num_cu, const std::st
         features[17] = num_cu;
         return features;
     }
-    std::size_t n = 18;
+    std::size_t n = 17;
     std::vector<float> features(n * n, 0.0f);
-    features[0]           = 2.0;
-    features[n + 1]       = problem.GetInChannels();
-    features[2 * n + 2]   = problem.GetInHeight();
-    features[3 * n + 3]   = problem.GetInWidth();
-    features[4 * n + 4]   = problem.GetOutChannels();
-    features[5 * n + 5]   = problem.GetOutHeight();
-    features[6 * n + 6]   = problem.GetOutWidth();
-    features[7 * n + 7]   = problem.GetWeightsHeight();
-    features[8 * n + 8]   = problem.GetWeightsWidth();
-    features[9 * n + 9]   = problem.GetPadH();
-    features[10 * n + 10] = problem.GetPadW();
-    features[11 * n + 11] = problem.GetKernelStrideH();
-    features[12 * n + 12] = problem.GetKernelStrideW();
-    features[13 * n + 13] = problem.GetDilationH();
-    features[14 * n + 14] = problem.GetDilationW();
-    features[15 * n + 15] = problem.GetBatchSize();
-    features[16 * n + 16] = problem.GetInDataType() == miopenFloat ? 2.0 : 1.0;
-    features[17 * n + 17] = problem.GetGroupCount();
+    features[0]           = problem.GetInChannels();
```

Comment:
why are we populating only the diagonal entries in an `n*n` matrix?

Reply:
That's the embedding method I created. This diagonal matrix scales embedding vectors that represent each convolution parameter with it's value.